src/geometry.jl

```diff
@@ -81,6 +82,130 @@ end
 
 @compat public revolution_volume
 push!(document[Symbol("Geometry")], :revolution_volume)
+"""
+    r_intersect_interval(x0::T, y0::T, dx::T, dy::T, r_min::T, r_max::T) where {T<:Real}
+
+Finds intersection of ray starting at x0 and y0 with direction (dx,dy) 
+    with reference major radius value r_ref
+
+- `x0``, `y0``: Origin
+- `dx``, `dy``: Normalized direction in x-y plane
+- `r_min`, `r_max`: Edges of r_range
+
+Returns interval [t1, t2] for which r_min <= r(t) <= r_max with r(t)^2 = (x+dx*t)^2 + (y+dy*t)^2
+
+"""
+
+function solve_r_intersect(x0::T, y0::T, dx::T, dy::T, r_min::T, r_max::T) where {T<:Real}
+    r0 = sqrt(x0^2 + y0^2)
+    t_crossings = zeros(Float64, 4)
+    crossing = zeros(Float64, 4) # +1 -> into domain
+                                 # -1 -> out of 
+                                 # 0 -> no crossing
+    into_domain = [1.0, -1.0]
+    if dx == 0 && dy == 0
+        # Handle vertical ray
+        if (r0 > r_min && r0 < r_max)
+            return [[0.0 Inf];]
+        else
+            return zeros(Float64, 0)
```

Comment:
You want an empty array? `Float64[]` is more idiomatic.

Reply:
```suggestion
            return Float64[]
```